src/txn/txn_ckpt.c

```diff
@@ -828,6 +828,9 @@ __checkpoint_prepare(WT_SESSION_IMPL *session, bool *trackingp, const char *cfg[
 
     __wt_writeunlock(session, &txn_global->rwlock);
 
+    /* Wait for the commit generation to drain before bumping the snapshot. */
+    __wt_gen_next_drain(session, WT_GEN_TXN_COMMIT);
```

Comment:
Can we run some perf tests to see how it impacts the performance? I suspect it may lead to perf regressions.

Reply:
Thanks @quchenhao. I have triggered couple of performance patch builds(WiredTiger & MongoDB) and updated the link in [WT-9199](https://jira.mongodb.org/browse/WT-9199)